apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java

```diff
@@ -111,6 +118,44 @@ public String deleteMultipleCollections() {
         return SUCCESS.toUpperCase();
     }
 
+    public static void dropSampleDataForApiCollection() {
+        List<ApiCollection> apiCollections = ApiCollectionsDao.instance.findAll(Filters.eq(ApiCollection.SAMPLE_COLLECTIONS_DROPPED, false));
+        if(apiCollections.isEmpty()) {
+            loggerMaker.infoAndAddToDb("No api collections to drop sample data for", LoggerMaker.LogDb.DASHBOARD);
+            return;
+        }
+        loggerMaker.infoAndAddToDb(String.format("Dropping sample data for %d api collections", apiCollections.size()), LoggerMaker.LogDb.DASHBOARD);
+        for (ApiCollection apiCollection: apiCollections) {
+            int apiCollectionId = apiCollection.getId();
+            loggerMaker.infoAndAddToDb("Dropping sti data for api collection: " + apiCollectionId, LoggerMaker.LogDb.DASHBOARD);
+            SingleTypeInfoDao.instance.deleteAll(Filters.eq("_id", Collections.singletonList(apiCollectionId)));
+            loggerMaker.infoAndAddToDb("Dropping sample data for api collection: " + apiCollectionId, LoggerMaker.LogDb.DASHBOARD);
+            SampleDataDao.instance.deleteAll(Filters.eq("_id.apiCollectionId", apiCollectionId));
+            loggerMaker.infoAndAddToDb("Dropping sensitive sample data for api collection: " + apiCollectionId, LoggerMaker.LogDb.DASHBOARD);
+            SensitiveSampleDataDao.instance.deleteAll(Filters.eq("_id.apiCollectionId", apiCollectionId));
+            ApiCollectionsDao.instance.updateOneNoUpsert(Filters.eq("_id", apiCollectionId), Updates.set(ApiCollection.SAMPLE_COLLECTIONS_DROPPED, true));
+        }
+        loggerMaker.infoAndAddToDb(String.format("Dropped sample data for %d api collections", apiCollections.size()), LoggerMaker.LogDb.DASHBOARD);
+    }
+
+    public String redactCollection() {
+        List<Bson> updates = Arrays.asList(
+            Updates.set(ApiCollection.REDACT, redacted),
+            Updates.set(ApiCollection.SAMPLE_COLLECTIONS_DROPPED, !redacted)
+        );
+        ApiCollectionsDao.instance.updateOneNoUpsert(Filters.eq("_id", apiCollectionId), Updates.combine(updates));
+        if(redacted){
+            int accountId = Context.accountId.get();
+            Runnable r = () -> {
+                Context.accountId.set(accountId);
+                loggerMaker.infoAndAddToDb("Triggered job to delete sample data", LoggerMaker.LogDb.DASHBOARD);
+                dropSampleDataForApiCollection();
```

Comment:
This function will redact data for all collections with sample_data_redacted == false. The API, I guess, wants to that for a singular collection.